packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -406,7 +404,9 @@ export class VpcNetwork extends VpcNetworkBase implements cdk.ITaggable {
 
     natSubnets = natSubnets.slice(0, natCount);
     for (const sub of natSubnets) {
-      this.natGatewayByAZ[sub.availabilityZone] = sub.addNatGateway();
+      const gateway = sub.addNatGateway();
+      this.natGatewayByAZ[sub.availabilityZone] = gateway.natGatewayId;
+      this.natDependencies.push(gateway);
```

Comment:
Wouldn’t it make more sense to just expose this as “natGateways” ?

Reply:
Maybe. But I don't want to overpromise; they were there to be able to take a dependency on, nothing more.

Reply:
Sounds like a legitimate API to expose the NAT gateways and internet gateways within the VPC, no?